src/main/java/com/example/java/maven/cardGamePeterPan/Game.java

```diff
@@ -47,59 +48,86 @@ public void printPlayers() {
-            return (userInput - 1);
+        try {
+            if (player.areCardsMatching(firstCardToDiscardIndex, secondCardToDiscardIndex)) {
+                player.discardCardsFromHand(firstCardToDiscardIndex, secondCardToDiscardIndex);
+            }
+            else {
+                messagePrinter.printError(WRONG_CARDS_TO_DISCARD);
+                choseCardsToDiscard(player);
+            }
         }
-        else {
-            messagePrinter.printError(String.format("Your input must be an integer between 1 and %s", player.getHand().size()));
-            return getValidCardIndex(player);
+        catch (IndexOutOfBoundsException exception) {
+            messagePrinter.printError(exception.getMessage());
+            choseCardsToDiscard(player);
         }
     }
 
     public void getCardFromPreviousPlayerHand(Player player) {
         Player previousPlayer = getPreviousPlayer(player);
-        messagePrinter.printMessage(String.format("%s has %s cards. Which one would you like to draw?", previousPlayer.getName(), previousPlayer.getHand().size()));
-        player.takeCardFromAnotherPlayersHand(previousPlayer, getValidCardIndex(previousPlayer));
+        messagePrinter.printMessage(String.format("%s has %s cards. Which one would you like to draw?", previousPlayer.getName(), previousPlayer.getHandSize()));
+        try {
+            player.takeCardFromAnotherPlayersHand(previousPlayer, userInputProvider.getNumberOfChosenCard());
+        }
+        catch (IndexOutOfBoundsException exception) {
+            messagePrinter.printError(exception.getMessage());
+            getCardFromPreviousPlayerHand(player);
+        }
     }
 
-
     public Player getPreviousPlayer(Player player) {
-        int indexOfCurrentPlayer = playerList.indexOf(player);
+        List<Player> activePlayers = getActivePlayers();
+        int indexOfCurrentPlayer = activePlayers.indexOf(player);
         if ((indexOfCurrentPlayer - 1) < 0) {
-            return playerList.get(playerList.size() - 1);
+            return activePlayers.get(activePlayers.size() - 1);
         }
         else {
-            return playerList.get(indexOfCurrentPlayer-1);
+            return activePlayers.get(indexOfCurrentPlayer-1);
         }
     }
 
     public int countActivePlayers() {
         int activePlayerCounter = 0;
         for (Player player : playerList) {
-            if (player.isPlaying()) {
+            if (player.getHandSize() > 0) {
                 activePlayerCounter += 1;
             }
         }
         return activePlayerCounter;
     }
+
+    public List<Player> getActivePlayers(){
+        return playerList.stream()
+                .filter(player -> player.getHandSize() > 0)
+                .collect(Collectors.toList());
+    }
+
+    public boolean checkIfPlayerIsActive(Player player) {
+        return player.getHandSize() != 0;
+    }
+
+    public void play() {
+        while (countActivePlayers() > 1) {
```

Comment:
when you are out of the loop you probably should print some message for the looser? :D

Reply:
To avoid situation when iteration continues while there is last player I added condition, that checks after every player if number of active players is greater than 1. If it equals 1, then last active player gets message about losing the game.